sklearn/model_selection/_validation.py

```diff
@@ -633,15 +654,34 @@ def _score(estimator, X_test, y_test, scorer):
                     # e.g. unwrap memmapped scalars
                     score = score.item()
             if not isinstance(score, numbers.Number):
-                raise ValueError(error_msg % (score, type(score), name))
-            scores[name] = score
+                if error_score == 'raise':
+                    raise ValueError(error_msg % (score, type(score), name))
+                elif isinstance(error_score, numbers.Number):
+                    scores[name] = error_score
+                    warnings.warn("Scoring failed. The score on this "
```

Comment:
I don't think we should replace the current error message... This is a problem of the API of the scorer, not its success at scoring.

Reply:
Hi @jnothman , do you mean we should still raise ValueError here when scoring failed irrespective of error_score (I mean, we shouldn't change score to error_score when scoring failed?)

Do you suggest we should handle the case only when MultiMetricScorer throws an exception?

Reply:
Yes, I would be inclined to still raise an error in this case, as closely the score is invalid, not the score

Reply:
ok, let me update it to change score to error_score only when MultiMetricScorer throws an exception

Reply:
Please review the latest patch.